core/domain/question_services.py

```diff
@@ -217,3 +226,141 @@ def update_question(
+    """
+    return question_models.QuestionModel.get_new_id('')
+
+
+def create_question_summary(question_id, creator_id):
+    """Creates and stores a summary of the given question.
+
+    Args:
+        question_id: str. ID of the question.
+        creator_id: str. The user ID of the creator of the question.
+    """
+    question = get_question_by_id(question_id, strict=False)
+    question_summary = compute_summary_of_question(question, creator_id)
+    save_question_summary(question_summary)
+
+
+def compute_summary_of_question(question, creator_id):
+    """Create a QuestionSummary domain object for a given Question domain
+    object and return it.
+
+    Args:
+        question: Question. The question object for which the summary
+            is to be computed.
+        creator_id: str. The user ID of the creator of the question.
+
+    Returns:
+        QuestionSummary. The computed summary for the given question.
+    """
+    question_html_data = question.question_data['content']['html']
+    question_summary = question_domain.QuestionSummary(
+        question.id, creator_id, question_html_data)
+
+    return question_summary
+
+
+def save_question_summary(question_summary):
+    """Save a question summary domain object as a QuestionSummaryModel
+    entity in the datastore.
+
+    Args:
+        question_summary: The question summary object to be saved in the
+            datastore.
+    """
+    question_summary_model = question_models.QuestionSummaryModel(
+        id=question_summary.id,
+        creator_id=question_summary.creator_id,
+        question_model_last_updated=question_summary.last_updated,
+        question_model_created_on=question_summary.created_on,
+        question_html_data=question_summary.question_html_data
+    )
+
+    question_summary_model.put()
+
+
+def get_question_summaries_by_creator_id(creator_id):
+    """Gets question summaries of questions created by the user.
+
+    Args:
+        creator_id: str. The user ID of the creator.
+
+    Returns:
+        QuestionSummaryModel. The QuestionSummaryModel for the given question.
+    """
+    return question_models.QuestionSummaryModel.get_by_creator_id(creator_id)
+
+
+def get_question_summary_by_question_id(question_id, strict=False):
+    """Gets question summaries of questions created by the user.
+
+    Args:
+        question_id: str. The user ID of the question.
+        strict: bool. Whether to fail noisily if no question summary for the
+            given question id exists in the datastore.
+
+    Returns:
+        QuestionSummaryModel. The QuestionSummaryModel for the given question.
+    """
+    return question_models.QuestionSummaryModel.get(question_id, strict=strict)
+
+
+def get_summaries_of_linked_skills(question_id):
+    """Gets linked skill IDs for given question.
+
+    Args:
+        question_id: str. The question ID for the given question.
+
+    Returns:
+        QuestionSkillLinkModel|None. The QuestionSkillModel for the given
```

Comment:
Never return storage models directly. Translate into domain objects first.

Even better, return the simplest possible thing. In this case it would be skill_id. Some other function can handle getting the summaries.

Reply:
Done.